docs/fundamentals/code-analysis/rules/quality-rules-reference.md

```diff
@@ -0,0 +1,281 @@
+> | [CA5394 Do not use insecure randomness](ca5394.md) | Using a cryptographically weak pseudo-random number generator may allow an attacker to predict what security-sensitive value will be generated. |
+> | [CA5395 Miss HttpVerb attribute for action methods](ca5395.md) | All the action methods that create, edit, delete, or otherwise modify data needs to be protected with the antiforgery attribute from cross-site request forgery attacks. Performing a GET operation should be a safe operation that has no side effects and doesn't modify your persisted data. |
+> | [CA5396 Set HttpOnly to true for HttpCookie](ca5396.md) | As a defense in depth measure, ensure security sensitive HTTP cookies are marked as HttpOnly. This indicates web browsers should disallow scripts from accessing the cookies. Injected malicious scripts are a common way of stealing cookies. |
+> | [CA5397: Do not use deprecated SslProtocols values](ca5397.md) | Transport Layer Security (TLS) secures communication between computers, most commonly with Hypertext Transfer Protocol Secure (HTTPS). Older protocol versions of TLS are less secure than TLS 1.2 and TLS 1.3 and are more likely to have new vulnerabilities. Avoid older protocol versions to minimize risk. |
+> | [CA5398: Avoid hardcoded SslProtocols values](ca5398.md) | Transport Layer Security (TLS) secures communication between computers, most commonly with Hypertext Transfer Protocol Secure (HTTPS). Protocol versions TLS 1.0 and TLS 1.1 are deprecated, while TLS 1.2 and TLS 1.3 are current. In the future, TLS 1.2 and TLS 1.3 may be deprecated. To ensure that your application remains secure, avoid hardcoding a protocol version. |
+> | [CA5399 Definitely disable HttpClient certificate revocation list check](ca5399.md) | A revoked certificate isn't trusted anymore. It could be used by attackers passing some malicious data or stealing sensitive data in HTTPS communication. |
+> | [CA5400 Ensure HttpClient certificate revocation list check is not disabled](ca5400.md) | A revoked certificate isn't trusted anymore. It could be used by attackers passing some malicious data or stealing sensitive data in HTTPS communication. |
+> | [CA5401 Do not use CreateEncryptor with non-default IV](ca5401.md) | Symmetric encryption should always use a non-repeatable initialization vector to prevent dictionary attacks. |
+> | [CA5402 Use CreateEncryptor with the default IV](ca5402.md) | Symmetric encryption should always use a non-repeatable initialization vector to prevent dictionary attacks. |
+> | [CA5403: Do not hard-code certificate](ca5403.md) | The `data` or `rawData` parameter of a <xref:System.Security.Cryptography.X509Certificates.X509Certificate> or <xref:System.Security.Cryptography.X509Certificates.X509Certificate2> constructor is hard-coded. |
+> | [IL3000 Avoid accessing Assembly file path when publishing as a single file](il3000.md) | Avoid using accessing Assembly file path when publishing as a single file. |
```

Comment:
Are IL3000 and IL3001 supposed to have the same description?

Reply:
@agocke